### PR TITLE
LLD TypeScript: solving some tradeoffs from the past

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

this PR will ship extra improvements we could achieve on the LLD TypeScript rework.

### ❓ Context

- **Impacted projects**: `lld` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/icons/device/interactions/NanoS/Swivel.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ type Props = {
   angle: number;
 };
 const NanoSSwivel = ({ angle, ...props }: Props) => {
-  const type = useTheme("colors.palette.type") as keyof typeof colors;
+  const type = useTheme().colors.palette.type as keyof typeof colors;
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/icons/device/interactions/NanoS/Swivel.tsx:13]: @mcayuelas-ledger do we still need the casting part?

File: apps/ledger-live-desktop/src/renderer/index.html [hunk 2]
```diff
@@ -173,12 +157,12 @@
...
+        V190.1z M0,479.3v190.1h289.2v-42.1H42.1V479.3H0z M0,0v190.1h42.1V42.1h247.1V0H0z" />
+        </svg>
       </div>
     </div>
-    <div id="modals"></div>
-  </body>
+  </div>
+  <div id="modals"></div>
+</body>
+
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/index.html:167]: @elbywan would it be possible not to regen the prettier in context of this PR? 🙏

File: apps/ledger-live-desktop/tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "../../tsconfig.base",
   "compilerOptions": {
+    "allowJs": false,
```
Comment (gre) [inline apps/ledger-live-desktop/tsconfig.json:4]: nice!

File: apps/ledger-live-desktop/src/renderer/index.html (hunk 2 above)
Comment (elbywan) [reply apps/ledger-live-desktop/src/renderer/index.html:167]: woops yeah I'll fix that 👍

File: apps/ledger-live-desktop/src/renderer/icons/device/interactions/NanoS/Swivel.tsx (hunk 1 above)
Comment (mcayuelas-ledger) [reply apps/ledger-live-desktop/src/renderer/icons/device/interactions/NanoS/Swivel.tsx:13]: Nope ! I handle it :)

**Pull request merged**